### PR TITLE
feat(synapse-interface): validate rfq quotes

### PR DESCRIPTION
**Pull request by bigboydiamonds**
null
9c76e8597298ba371a50b300d41208a66d2d25d5: [synapse-interface preview link ](https://sanguine-synapse-interface-lmkf4dlk6-synapsecns.vercel.app)
0fe57cd125e21f6c31fdd079a7ad45ea8a7e4223: [synapse-interface preview link ](https://sanguine-synapse-interface-clsplbzwe-synapsecns.vercel.app)
9bc4102f1ac2751ad80ef1efd4e648c51423f604: [synapse-interface preview link ](https://sanguine-synapse-interface-gh0vy2guc-synapsecns.vercel.app)
5c7b8d351dbf01d6fd34c61263b2dea87afcb870: [synapse-interface preview link ](https://sanguine-synapse-interface-e5k0qbs4j-synapsecns.vercel.app)
35b1df06dc851122147a30a391a47a1743fa2b9b: [synapse-interface preview link ](https://sanguine-synapse-interface-5mgx890xl-synapsecns.vercel.app)
9aff122b8ea64dcd754b495833aabe6ff1c3edbc: [synapse-interface preview link ](https://sanguine-synapse-interface-da5wg2bmq-synapsecns.vercel.app)
7b6206b96bdc4b2893c7f930ea692b8ff0a7471e: [synapse-interface preview link ](https://sanguine-synapse-interface-4mpur5az5-synapsecns.vercel.app)

## Summary by CodeRabbit

- **New Features**
	- Enhanced decision-making process for quote selection in the State Managed Bridge component.

- **Improvements**
	- Simplified analytics event tracking by removing user account address, focusing on chain IDs and token details.

- **Chores**
	- Streamlined event logging across multiple components by excluding the address from analytics.

d1e3f29e9918104a861ba1cdb1d85b0c7c60fe57: [synapse-interface preview link ](https://sanguine-synapse-interface-f14lkxyd3-synapsecns.vercel.app)
8081ae0f7513440aa1813468fe9606bc4c10ec9f: [synapse-interface preview link ](https://sanguine-synapse-interface-10vsxblkn-synapsecns.vercel.app)
beb82df5d9c93e43597e58ddb15d44ed37ac617a: [synapse-interface preview link ](https://sanguine-synapse-interface-cmawo2mu4-synapsecns.vercel.app)

### CODE REVIEW

**Review by ChiTimesChi — APPROVED** [review]
LGTM with a few minor nits

File: packages/synapse-interface/pages/state-managed-bridge/index.tsx [hunk 1]
```diff
@@ -187,12 +187,38 @@ const StateManagedBridge = () => {
...
+            destinationTokenAddress: toToken.addresses[toChainId],
+            rfqQuoteAmountOut: rfqQuote.maxAmountOut.toString(),
+            nonRfqMaxAmountOut: nonRfqQuote.maxAmountOut.toString(),
+          })
+        }
+      } else if (rfqQuote) {
         quote = rfqQuote
       } else {
-        /* allBridgeQuotes returns sorted quotes by maxAmountOut descending */
         quote = activeQuotes[0]
```
[the remark this answers is not in the document]
Comment (bigboydiamonds) [reply packages/synapse-interface/pages/state-managed-bridge/index.tsx]: Great catch! That's cleaner logic, updated in https://github.com/synapsecns/sanguine/pull/2971/commits/8c1e247628a8cea228779746177a497ea22936bf

File: packages/synapse-interface/pages/state-managed-bridge/index.tsx [hunk 2]
```diff
@@ -187,12 +187,38 @@ const StateManagedBridge = () => {
...
+      )
+
       let quote
 
-      if (rfqQuote) {
+      if (rfqQuote && nonRfqQuote) {
+        const rfqMaxAmountOut = BigInt(rfqQuote.maxAmountOut.toString())
+        const nonRfqMaxAmountOut = BigInt(nonRfqQuote.maxAmountOut.toString())
+
+        const maxDifference = (nonRfqMaxAmountOut * 30n) / 100n
```
[the remark this answers is not in the document]
Comment (bigboydiamonds) [reply packages/synapse-interface/pages/state-managed-bridge/index.tsx]: Assigned constant name `allowedPercentileDifference` - do you prefer a different name?

[the remark this answers is not in the document]
Comment (bigboydiamonds) [reply packages/synapse-interface/pages/state-managed-bridge/index.tsx]: got it! updated in https://github.com/synapsecns/sanguine/pull/2971/commits/ebf52df13a659fcf21636345181cb34613160ac6

**Pull request merged**